eisbuk-admin/src/containers/SlotsPageContainer.tsx

```diff
@@ -184,4 +167,22 @@ const SlotsPageContainer: React.FC<Props> = ({ slots, children, ...props }) => {
   );
 };
 
+/**
+ * Generates dates to display based on the view:
+ * - 7 days of the week in case of ice
+ * - 4-5 Mondays, Tuesdays, etc. of the month in case of off-ice
+ * @param currentDate start date for current view
+ * @param view passed from props: "book_ice", "book_off_ice" or "ice"
+ * @returns array of dates to display
+ */
+const getDatesToDisplay = (currentDate: DateTime, view?: CustomerRoute) =>
+  view === CustomerRoute.BookIce
+    ? Array(5)
+        .fill(null)
+        .map((_, i) => currentDate.plus({ week: i }).toISODate())
+        .filter((date) => DateTime.fromISO(date).hasSame(currentDate, "month"))
+    : Array(7)
+        .fill(null)
+        .map((_, i) => currentDate.plus({ days: i }).toISODate());
+
```

Comment:
In my opinion, it's better to write the getter or calculating functions outside of the component. This is good idea as long as the function doesn't do some state change or interact with the hooks. So please, take a look at what I did here.
This both improves readability (by not overcluttering the component code) and has some added benefits because you don't need to memoize the function as it's constant outside of the component. 
*If you want more info on `React.FC` rendering and rerendering flow and memoization let me know as it's a lengthy subject.

Reply:
Actually I did have something almost exactly like this (that's why I wrote JSDocs for it), later I felt like a function was overkill cause it's basically one argument and it's a global prop so I removed the function but forgot to remove the docs.
I didn't take into consideration the fact that it would be better outside the component so that's why I felt like the function was unnecessary, but yeah I see the point.